hack/helm/release/internal/helm/helm.go

```diff
@@ -0,0 +1,616 @@
+// to a given chartPath temporary directory.
+func copyChartToTemporaryDirectory(ch chart, chartPath string) error {
+	err := os.Mkdir(chartPath, 0755)
+	if err != nil {
+		return fmt.Errorf("while making chart directory inside temporary directory (%s): %w", chartPath, err)
+	}
+	err = copy(ch.fullPath, chartPath)
+	if err != nil {
+		return fmt.Errorf("while copying chart (%s) to temporary directory: %w", ch.Name, err)
+	}
+	return nil
+}
+
+// runChartDependencyUpdate runs 'helm dependency update chart' for a given Helm chart.
+func runChartDependencyUpdate(chartPath, chartName string) error {
+	settings := cli.New()
+	client := action.NewDependency()
+	cfg := action.Configuration{}
+	man := &downloader.Manager{
+		Out:              os.Stdout,
+		ChartPath:        chartPath,
+		Keyring:          client.Keyring,
+		SkipUpdate:       client.SkipRefresh,
+		Getters:          getter.All(settings),
+		RegistryClient:   cfg.RegistryClient,
+		RepositoryConfig: settings.RepositoryConfig,
+		RepositoryCache:  settings.RepositoryCache,
+		Debug:            settings.Debug,
+		Verify:           downloader.VerifyNever,
+	}
+	if client.Verify {
+		man.Verify = downloader.VerifyAlways
+	}
+	log.Printf("Updating dependencies for chart (%s)\n", chartName)
+	if err := man.Update(); err != nil {
+		return fmt.Errorf("while updating dependencies for helm chart (%s): %w", chartName, err)
+	}
+	return nil
+}
+
+// packageHelmChart runs 'helm package chart' for a given Helm chart.
+func packageHelmChart(chartPath, chartName string) error {
+	packageClient := action.NewPackage()
+	log.Printf("Packaging chart (%s)\n", chartName)
+	if _, err := packageClient.Run(chartPath, map[string]interface{}{}); err != nil {
+		return fmt.Errorf("while packaging helm chart (%s): %w", chartName, err)
+	}
+	return nil
+}
+
+// copyChartToBucketConfig is the configuration for copying
+// a chart to a GCS bucket.
+type copyChartToBucketConfig struct {
+	bucket, chartName, tempDirectory, sourceGlob string
+	dryRun                                       bool
+}
+
+// copyChartToGCSBucket will potentially copy a Helm chart to a GCS bucket.
+// If the object already exists within the bucket, it will not be overwritten.
+func copyChartToGCSBucket(ctx context.Context, config copyChartToBucketConfig) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+
+	storageClient, err := storage.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("while creating gcs storage client: %w", err)
+	}
+	defer storageClient.Close()
+
+	files, err := filepath.Glob(config.sourceGlob)
+	if err != nil {
+		return fmt.Errorf("while search for file glob (%s): %w", config.sourceGlob, err)
+	}
+	if len(files) == 0 {
+		return fmt.Errorf("couldn't file helm package with glob (%s)", config.sourceGlob)
+	}
+	// TODO is this 'helm' prefix always assumed?  I don't think so...
+	// This should be the trailing path of the given repo url, right?
```

Comment:
Can we check this and remove the TODO?

Reply:
Thanks for pointing this out.  I've verified, and since we're using constants now for dev/prod, we can assume we can use the `Path` parsed from "net/url".Parse, with the leading `/` trimmed.  👍